File: Trabajo_7/JAVA/Ejercicios/Ejercicio_3.java
package Trabajo_7.Ejercicios;

class Movil {
    private String marca;
    private String modelo;
    private String sistema;

    public Movil(String marca, String modelo, String sistema) {
        this.marca = marca;
        this.modelo = modelo;
        this.sistema = sistema;
    }

    public String getMarca() {
        return marca;
    }

    public String getModelo() {
        return modelo;
    }

    public String getSistema() {
        return sistema;
    }
}

class Telefono extends Movil {
    private int nCelular;

    public Telefono(String marca, String modelo, String sistema, int nCelular) {
        super(marca, modelo, sistema);
        this.nCelular = nCelular;
    }

    @Override
    public String toString() {
        return "Datos del telefono:\n" +
                "Marca             -> " + getMarca() + "\n" +
                "Modelo            -> " + getModelo() + "\n" +
                "Sistema Operativo -> " + getSistema() + "\n" +
                "Numero Celular    -> " + nCelular;
    }
}

class Tablet extends Movil {
    private boolean chip;

    public Tablet(String marca, String modelo, String sistema, boolean chip) {
        super(marca, modelo, sistema);
        this.chip = chip;
    }

    @Override
    public String toString() {
        return "Datos de la Tablet:\n" +
                "Marca             -> " + getMarca() + "\n" +
                "Modelo            -> " + getModelo() + "\n" +
                "Sistema Operativo -> " + getSistema() + "\n" +
                "Tiene chip        -> " + (chip ? "Si" : "No");
    }
}



public class Ejercicio_3 {
    public static void main(String[] args) {
        Telefono tel01 = new Telefono("Samsung", "A150", "Linux", 123456789);
        Tablet tab01 = new Tablet("IPhone", "90", "Ubuntu", true);
        System.out.println(tel01);
        System.out.println();
        System.out.println(tab01);
    }
}

File: Trabajo_7/CPP/Ejercicios/Ejercicio_3.cpp
#include <iostream>
using namespace std;

class Movil{
private:
    string marca;
    string modelo;
    string sistema;

public:
    Movil(string brand, string model, string system){
        this -> marca = brand;
        this -> modelo = model;
        this -> sistema = system;
    }

    ~Movil(){};

    //Metodos G y S
    string getMarca() {
        return marca;
    }

    string getModelo() {
        return modelo;
    }

    string getSistema() {
        return sistema;
    }

};

class Telefono : public Movil{
private:
    int n_celular;
public:
    Telefono(string brand, string model, string system, int n_cel)
            : Movil(brand, model, system) {
        this -> n_celular = n_cel;
    }

    ~Telefono(){
        cout << "Destruido telefono con N celular -> " << this -> n_celular << endl;
    };

    friend ostream& operator << (ostream& os, Telefono& Tel_01){
        os << "Datos del telefono:" << endl;
        os << "Marca             -> " << Tel_01.getMarca() << endl;
        os << "Modelo            -> " << Tel_01.getModelo() << endl;
        os << "Sistema Operativo -> " << Tel_01.getSistema() << endl;
        os << "Numero Celular    -> " << Tel_01.n_celular << endl;

        return os;
    }



};

class Tablet : public Movil{
private:
    bool chip;
public:
    Tablet(string brand, string model, string system, bool chap)
            : Movil(brand, model, system) {
        this -> chip = chap;
    }

    ~Tablet(){
        cout << "Destruido tablet con Modelo -> " << this -> getModelo() << endl;
    };

    friend ostream& operator << (ostream& os, Tablet& Tab_01){
        os << "Datos de la Tablet:" << endl;
        os << "Marca             -> " << Tab_01.getMarca() << endl;
        os << "Modelo            -> " << Tab_01.getModelo() << endl;
        os << "Sistema Operativo -> " << Tab_01.getSistema() << endl;
        if (Tab_01.chip == false){
            os << "Tiene chip        -> Si" << endl;
        } else{
            os << "Tiene chip        -> No" << endl;
        }

        return os;
    }



};


int main(){
    Telefono Tel_01("Samsung", "A150", "Linux", 123456789);
    Tablet Tab_01("IPhone", "90", "Ubuntu", 1);
    cout << Tel_01;
    cout << "" << endl;
    cout << Tab_01;
    cout << "" << endl;

    return 0;
}